searchv2/builders_v3.py

```diff
@@ -0,0 +1,208 @@
+#             weight += 20
+#         elif now - last_updated < timedelta(182):
+#             weight += 10
+#         elif now - last_updated < timedelta(365):
+#             weight += 5
+
+#     return weight
+
+
+def index_packages(*, verbose: bool = False):
+    rules = [
+        DeprecatedRule(),
+        DescriptionRule(),
+        DownloadsRule(),
+        ForkRule(),
+        LastUpdatedRule(),
+        RecentReleaseRule(),
+        UsageCountRule(),
+        WatchersRule(),
+    ]
+
+    # package = Package.objects.first()
+
+    # demo the group rule
+
+    # group = ScoreRuleGroup(
+    #     name="Activity Rules",
+    #     description="Rules related to the package's recent activity",
+    #     max_score=40,
+    #     documentation_url="https://docs.yoursite.com/rules/groups/activity",
+    #     rules=[LastUpdatedRule(), RecentReleaseRule()],
+    # )
+
+    # rules = [
+    #     DeprecatedRule(),
+    #     DescriptionRule(),
+    #     DownloadsRule(),
+    #     ForkRule(),
+    #     UsageCountRule(),
+    #     WatchersRule(),
+    #     group,
+    # ]
+
+    # package_score = calc_package_weight(package=package, rules=rules, max_score=100)
+
+    # print(json.dumps(package_score, indent=2))
+
+    for package in Package.objects.all().iterator():
+        package_score = calc_package_weight(package=package, rules=rules, max_score=100)
+        weight = package_score["total_score"]
+
+        if verbose:
+            print(f"{package.pk=}::{weight=}")
+            print(json.dumps(package_score, indent=2))
+
+        obj, created = SearchV2.objects.update_or_create(
+            item_type="package",
+            slug=package.slug,
+            defaults={
+                "absolute_url": package.get_absolute_url(),
+                "category": package.category.title,
+                "clean_title": clean_title(remove_prefix(package.slug)),
+                "description": package.repo_description,
+                "participants": package.participants,
+                "pypi_downloads": package.pypi_downloads,
+                "repo_forks": package.repo_forks,
+                "repo_watchers": package.repo_watchers,
+                "slug_no_prefix": remove_prefix(package.slug),
+                "title": package.title,
+                "title_no_prefix": remove_prefix(package.title),
+                "usage": package.usage.count(),
+                "weight": weight,
+            },
+        )
+
+        optional_save = False
+        try:
+            obj.last_committed = package.last_updated()
+            optional_save = True
+        except Commit.DoesNotExist:
+            pass
+
+        last_released = package.last_released()
+        if last_released and last_released.upload_time:
+            obj.last_released = last_released.upload_time
+            optional_save = True
+
+        if optional_save:
+            obj.save()
+
+
+def calc_grid_weight(
+    *,
+    grid: Grid,
+    max_weight: int = 120,
+) -> int:
```

Comment:
does it seem the linter got a bit aggressive here. It just looks weird to me, but 🤷

Reply:
@ryancheley Yes, I love it, though. Give me pull request friendly diffs over one-liners. 😋